src/components/ModalWithConfirmation.js

```diff
@@ -0,0 +1,37 @@
+import Modal from "./Modal.js";
+
+export default class ModalWithConfirmation extends Modal {
+  constructor(modalSelector, { submitButtonSelector, formSelector }) {
+    super(modalSelector);
+    this._button = this._modalElement.querySelector(submitButtonSelector);
+    this._callback = null;
+    this._form = this._modalElement.querySelector(formSelector);
+    this._originalButtonText = this._button.textContent;
+  }
+
+
+  renderingSaving(isSaving) {
```

Comment:
Great that you've implemented the logic for rendering loading via the class methods!